bluemira/power_cycle/net.py

```diff
@@ -0,0 +1,648 @@
+
+
+@dataclass
+class SubSystemConfig(Config):
+    """Power cycle sub system config"""
+
+    loads: List[str]
+    description: str = ""
+
+
+@dataclass
+class LoadConfig(Config):
+    """Power cycle load config"""
+
+    time: npt.ArrayLike = field(default_factory=lambda: np.arange(2))
+    reactive_data: Optional[npt.ArrayLike] = None
+    active_data: Optional[npt.ArrayLike] = None
+    efficiencies: LoadEfficiencyDescriptor = LoadEfficiencyDescriptor()
+    model: Union[LoadModel, str] = LoadModel.RAMP
+    unit: str = "W"
+    description: str = ""
+    normalised: bool = True
+    consumption: bool = True
+
+    def __post_init__(self):
+        """Validate load"""
+        if self.reactive_data is None:
+            self.reactive_data = np.zeros_like(self.time)
+        if self.active_data is None:
+            self.active_data = np.zeros_like(self.time)
+
+        for var_name in ("time", "reactive_data", "active_data"):
+            var = getattr(self, var_name)
+            if not isinstance(var, np.ndarray):
+                setattr(self, var_name, np.asarray(var))
+        if isinstance(self.model, str):
+            self.model = LoadModel[self.model.upper()]
+        for data in (self.reactive_data, self.active_data):
+            if data.size != self.time.size:
+                raise ValueError(
+                    f"time and data must be the same length for {self.name}: {data}"
+                )
+        if any(np.diff(self.time) < 0):
+            raise ValueError("time must increase")
+
+        self.reactive_data = raw_uc(self.reactive_data, self.unit, "W")
+        self.active_data = raw_uc(self.active_data, self.unit, "W")
+        self.unit = "W"
+
+    def interpolate(
+        self,
+        time: npt.ArrayLike,
+        end_time: Optional[float] = None,
+        load_type: Union[str, LoadType] = LoadType.ACTIVE,
+    ) -> np.ndarray:
+        """
+        Interpolate load for a given time vector
+
+        Notes
+        -----
+        The interpolation type is set by load.model.
+        Any out-of-bound values are set to zero.
+        """
+        if isinstance(load_type, str):
+            load_type = LoadType[load_type.upper()]
+        return interp1d(
+            self.time,
+            getattr(self, f"{load_type.name.lower()}_data"),
+            kind=self.model.value,
+            bounds_error=False,  # turn-off error for out-of-bound
+            fill_value=(0, 0),  # below-/above-bounds extrapolations
+        )(time if end_time is None else np.array(time) * end_time)
+
+
+def interpolate_extra(vector: npt.NDArray, n_points: int):
+    """
+    Add points between each point in a vector.
+    """
+    if n_points == 0:
+        return vector
+
+    return np.concatenate([
+        *(
+            np.linspace(vector[s], vector[s + 1], n_points + 1, endpoint=False)
+            for s in range(len(vector) - 1)
+        ),
+        np.atleast_1d(vector[-1]),
+    ])
+
+
+class Loads:
+    """Loads of a phase"""
```

Comment:
Discussed elsewhere summary:

If a user wants to add a load to the config they are interfacing with the config. They are not at the same time dealing with the resultant phase and loads on that phase.

If in future we do generalise its best to do it while creating that usecase instead of trying to make it too general too soon.

On the whole we shouldnt need to implement basic mathematical operations on classes. There needs to be a very good reason for this and adding simple arrays together on the surface doesnt seem to be one of them